File: web/webapp/src/enums/ErrorCode.js
export const ErrorCode = {
    ACCESS_DENIED: {
        code: 403,
        message: "ACCESS_DENIED",
        isLogout: true,
    },
    SESSION_EXPIRED: {
        code: 1003,
        message: "SESSION_EXPIRED",
        isLogout: true,
    },
    DISPLAY_ORDER_DUPLICATED: {
        code: 1002,
        message: "DISPLAY_ORDER_DUPLICATED",
    },
    GROUP_PREFERENCE_IN_USE: {
        code: 1011,
        message: "GROUP PREFERENCE IN USE",
    },
    resolve(code) {
        return Object.values(this).find(errorCode => errorCode.code === code);
    }
};

File: api/src/main/java/com/alim/allconnp/api/advice/ErrorCode.java
package com.alim.allconnp.api.advice;

public enum ErrorCode implements AbstractCode<ErrorCode> {
    BAD_REQUEST(400, "BAD REQUEST"),
    UNAUTHORIZED(401, "UNAUTHORIZED"),
    ACCESS_DENIED(403, "ACCESS DENIED"),
    SESSION_EXPIRED(512, "SESSION EXPIRED");

    ErrorCode(int code, String message) {
        this.code = code;
        this.message = message;
    }

    private final int code;
    private final String message;

    public int getCode() {
        return this.code;
    }

    @Override
    public String getMessage() {
        return message;
    }
}
